client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -46,6 +53,20 @@ object PathBlob {
     }
   }
 
+  /** Factory method to combine two PathBlob instances
+   *  @param firstBlob: first pathblob
+   *  @param secondBlob: second pathblob
+   *  @return failure if the two blobs have conflicting entries; the combined blob otherwise.
+   */
+  def apply(firstBlob: PathBlob, secondBlob: PathBlob): Try[PathBlob] = {
```

Comment:
Call this a more meaningful name, like `reduce`. This gets very confusing since it's mixing a reducer and a constructor.

Reply:
Fixed.